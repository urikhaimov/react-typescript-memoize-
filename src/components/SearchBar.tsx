import React from 'react';
import 'reactjs-popup/dist/index.css';


export interface Props {
    changeHandler(e: any): void;
}

const SearchBar: React.FC<Props> = ({ changeHandler }) => {
   

    return (

        <div className="search-bar">
           <label>Search
                <input id="searchbartext" onChange={changeHandler} />
            </label>
        </div>
    );
};

export default SearchBar;
