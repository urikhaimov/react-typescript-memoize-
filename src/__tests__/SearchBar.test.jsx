import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, fireEvent, render } from '@testing-library/react';
import SearchBar from '../components/SearchBar';
const filterCountries = jest.fn();
afterEach(cleanup);
test('render Search Bar', () => {
 
  const component = renderer.create(
    <SearchBar changeHandler={filterCountries} />,

  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();


});

it('Search Bar after click', () => {
  const { queryByLabelText, getByLabelText } = render(
    <SearchBar changeHandler={filterCountries} />,
  );
  const input = queryByLabelText('Search');
  expect(input).toBeTruthy();
  fireEvent.change(getByLabelText('Search'), {target: {value: 'wonder'}})
  expect(input.value).toEqual('wonder');
});