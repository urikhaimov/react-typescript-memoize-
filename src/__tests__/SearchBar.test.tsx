import React from 'react';
import renderer from 'react-test-renderer';
import SearchBar from '../components/SearchBar';


test('render Sender Search Bar', () => {
  const filterCountries = jest.fn()
  const component = renderer.create(
    <SearchBar changeHandler={filterCountries} />,

  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();


});