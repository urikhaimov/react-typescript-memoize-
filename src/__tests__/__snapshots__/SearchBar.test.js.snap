// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Search Bar 1`] = `
<div
  className="search-bar"
>
  <label>
    Search
    <input
      id="searchbartext"
      onChange={[MockFunction]}
    />
  </label>
</div>
`;
